spec/lib/audit/catalog_to_moab_class_spec.rb

```diff
@@ -26,6 +26,25 @@
       expect(c2m_mock).to receive(:check_catalog_version).exactly(2).times
       described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir)
     end
+    it 'checks a PreservedCopy previously audited before one that is not audited' do
```

Comment:
I think this is the opposite of what we want

Reply:
confusingly, i think it's the test name that's wrong here.  the code currently uses this for the order by:
`'last_version_audit IS NOT NULL, last_version_audit ASC'`

what i think that does is sort nulls first, then non-nulls in ascending order.  the logic being:
* the first sort criterion is: 'date not null?'  all rows where the date is null will be false by that criterion, and i believe false results sort above true results (0 < 1).
* the second sort criterion just the date in ascending order, so all non-null dates will be sorted next by the actual date value.

in oracle, you could just do `'last_version_audit ASC NULLS FIRST'`, but when i went looking around on the internet to see if that was standard SQL, i saw stack exchange posts and whatnot jumping through hoops because they said it wasn't.  didn't do any testing of my own to confirm whether they were right.

Reply:
just did some manual testing in rails console, the kind of `order by` statement used in this PR indeed sorted nulls first for me.

Reply:
again, thanks for the details.  I think we want tests that explicitly show

- null value for `last_version_audit` gets done before a dated value
- dates come in oldest -> newest order
- dates AFTER the date arg are not included

Reply:
based on how completely un-intuitive this construction is, though, a comment might be _in order_ 😆

Reply:
ow ow ow

Reply:
oh wow, I got that just now! slow burn!